File: java/01-commands/doc/sourcecode/src/main/java/com/training/app/Account.java
package com.training.app;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Account {
    private int balance;
    private List<BankingTransactionDTO> statement;

    public Account() {
        statement = new ArrayList<>();
    }

    public int getBalance() {
        return balance;
    }

    public List<BankingTransactionDTO> getStatement() {
        return statement;
    }

    public void deposit(int depositAmount) {
        balance += depositAmount;
        statement.add(new BankingTransactionDTO(LocalDate.now(), depositAmount, balance));
    }

    public void withdraw(int withdrawAmount) {
        balance -= withdrawAmount;
        statement.add(new BankingTransactionDTO(LocalDate.now(), -withdrawAmount, balance));
    }
}

File: nodejs/01-commands/doc/sourcecode/Account.js
const BankingTransactionDTO = require('./BankingTransactionDTO');

class Account {
    constructor() {
        this.balance = 0;
        this.statement = [];
    }

    getBalance() {
        return this.balance;
    }

    getStatement() {
        return this.statement;
    }

    deposit(depositAmount) {
        this.balance += depositAmount;
        this.statement.push(new BankingTransactionDTO(new Date(), depositAmount, this.balance));
    }
}

module.exports = Account;